WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift

```diff
@@ -198,18 +199,26 @@ final class OrderListViewController: UIViewController, GhostableViewController {
         //
         // We can remove this once we've replaced XLPagerTabStrip.
         tableView.reloadData()
-
-        // Select the first order if we're showing in an open split view (i.e. on iPad in some size classes)
-        guard let splitViewController,
-              !splitViewController.isCollapsed else {
-            return
-        }
-        checkSelectedItem()
     }
 
     override func viewDidLayoutSubviews() {
         super.viewDidLayoutSubviews()
         tableView.updateHeaderHeight()
+
+        // The original implementation is to check the selected item in `viewWillAppear` when split view is enabled,
+        // but `UISplitViewController.isCollapsed` can be `false` in `viewWillAppear` in compact devices on certain iOS versions
+        // like iOS 17.0.1 and 17.1.
+        // To fix this issue, the selected item checking is now called once after the first `viewDidLayoutSubviews` where `isCollapsed` value is
+        // correctly set.
+        if !didCheckSelectedItemInSplitView {
+            didCheckSelectedItemInSplitView = true
+            // Select the first order if we're showing in an open split view (i.e. on iPad in some size classes)
+            guard let splitViewController,
+                  !splitViewController.isCollapsed else {
+                return
+            }
+            checkSelectedItem()
+        }
```

Comment:
```suggestion
        // To fix this issue, the selected item checking is now called after `viewDidLayoutSubviews`, where `isCollapsed` value is
        // correctly set.
        // This additionally ensures that an order is selected when changing from horizontally compact to regular.
        // Select the first order if we're showing in an open split view (i.e. on iPad in some size classes)
        guard let splitViewController,
              !splitViewController.isCollapsed else {
            return
        }
        checkSelectedItem(
```

Not sure about this, so please test it beyond what I've done, but I don't think we need the flag.

Removing it also fixes #11675. However, it might more properly be fixed by adding a call to `checkSelectedItem` to `OrderListViewController.willTransition(to:with:)`.

`willTransition` is called before `viewDidLayoutSubviews`, but both are called on trait change.